crates/indexer/src/market/service/execution.rs

```diff
@@ -0,0 +1,383 @@
+// Copyright 2025 Boundless Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crate::db::market::{CycleCountExecution, CycleCountExecutionUpdate};
+use crate::db::{DbObj, IndexerDb};
+use crate::market::service::IndexerServiceExecutionConfig;
+use alloy::primitives::{B256, U256};
+use anyhow::Result;
+use bonsai_sdk::non_blocking::{Client as BonsaiClient, SessionId};
+use boundless_market::storage::fetch_url;
+use broker::futures_retry::retry;
+use bytes::Bytes;
+use std::collections::HashSet;
+
+pub async fn execute_requests(db: DbObj, config: IndexerServiceExecutionConfig) {
+    tracing::info!("Started execution task");
+
+    const PENDING_LIMIT: u32 = 10;
+    const EXECUTING_LIMIT: u32 = 40;
+
+    let mut interval = tokio::time::interval(config.execution_interval);
+
+    let bento_client = BonsaiClient::from_parts(
+        config.bento_api_url.clone().unwrap(),
+        config.bento_api_key.clone().unwrap(),
+        risc0_zkvm::VERSION,
+    )
+    .unwrap();
+
+    loop {
+        interval.tick().await;
+
+        // Find cycle counts still in state PENDING
+        tracing::debug!("Querying DB for cycle counts in status PENDING...");
```

Comment:
It would be useful to log the size of the queue here so we can easily see how far behind we are, e.g. add a `SELECT COUNT(*) ... where PENDING` query on the table and have a log like `Currently X orders pending cycle counts, Y orders currently executing`

Reply:
Added a function to count rows in status PENDING, EXECUTING and FAILED, used for logging currently but could be useful later for other logic. https://github.com/boundless-xyz/boundless/pull/1441/commits/311671185d41421effdcf5004e4f97a01453ba52